src/mds/MDSRank.cc

```diff
@@ -2230,6 +2230,31 @@ bool MDSRank::command_dirfrag_ls(
   return true;
 }
 
+void MDSRank::dump_status(Formatter *f)
+{
+  Mutex::Locker l(mds_lock);
```

Comment:
MDSRank::mds_lock is a reference to MDSDaemon::mds_lock. Maybe we should take this lock at the start of MDSDaemon::dump_status rather than here.